alot/widgets/ansi.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2011-2012  Patrick Totzke <patricktotzke@gmail.com>
+# This file is released under the GNU GPL, version 3 or a later revision.
+# For further details see the COPYING file
+
+import urwid
+
+
+class ANSIText(urwid.WidgetWrap):
+
+    """Selectable Text widget that interprets ANSI color codes"""
+
+    def __init__(self, txt,
+                 default_attr=None,
+                 default_attr_focus=None,
+                 ansi_background=True, **kwds):
+        ct, focus_map = parse_escapes_to_urwid(txt, default_attr,
+                                               default_attr_focus,
+                                               ansi_background)
+        t = urwid.Text(ct, **kwds)
+        attr_map = {default_attr.background: ''}
+        w = urwid.AttrMap(t, attr_map, focus_map)
+        urwid.WidgetWrap.__init__(self, w)
+
+    def selectable(self):
+        return True
+
+    def keypress(self, size, key):
+        return key
+
+
+def parse_escapes_to_urwid(text, default_attr=None, default_attr_focus=None,
+                           parse_background=True):
+    """This function converts a text with ANSI escape for terminal
+    attributes and returns a list containing each part of text and its
+    corresponding Urwid Attributes object, it also returns a dictionary which
+    maps all attributes applied here to focused attribute.
+    """
+    ECODES = {
```

Comment:
For performance reasons I would not put this dict in the function, if we put it at the module level it will be initialized once, inside the function it will be re-initialized each time the function is called.